calculators/_memoria_matchups.php

```diff
@@ -20,38 +21,45 @@ function _memoria_matchups__main(&$teams) {
     if (count($teams[0]) !== count($teams[1])) {
         return;
     }
-    
+
     $game_duration_seconds = floor($teams[0][0]['duration'] / 22.4);
     if ($game_duration_seconds < 240) {
         return;
     }
-    
+
     $factor = 4000;
     $matchup_predictions = [];
     $matchup_predictions_weight = [];
     $matchup_predictions_weight_total = 0;
     for ($i = 0; $i < count($teams[0]); $i++) {
         $elo_part = abs(($teams[0][$i]['rating'] - $teams[1][$i]['rating']) / $factor);
         $matchup_predictions[$i] = floor(1000 / (1 + 10 ** $elo_part)) / 1000;
-		
+        if (count($teams[0]) > 1) {
+            $matchup_predictions[$i] -= $matchup_predictions[$i] ** (2 / (0.05 + $elo_part));
+            if ($matchup_predictions[$i] < 0) {
+                $matchup_predictions[$i] = 0;
+            }
+        }
+
         if ($teams[0][$i]['rating'] > $teams[1][$i]['rating']) {
             $matchup_predictions[$i] = 1 - $matchup_predictions[$i];
         }
-        
-        $matchup_predictions_weight[$i] = ((abs(0.5 - $matchup_predictions[$i]) + 0.015) * 6) ** ($elo_part * 1.5);
+
+        $matchup_predictions_weight[$i] = ((abs(0.5 - $matchup_predictions[$i]) + 1) * 3) ** ($elo_part / 1.25 + (max($teams[0][$i]['rating'], $teams[1][$i]['rating']) / $factor));
```

Comment:
`(max($teams[0][$i]['rating'], $teams[1][$i]['rating']) / $factor)` extra parentheses are fine for readability, but +- and */ operations priority are very clear, do not require extra parentheses.

To visually guide the reader, you may omit the spaces around / here, `** ($elo_part/1.25 + max($teams[0][$i]['rating'], $teams[1][$i]['rating'])/$factor)`